addon/components/polaris-badge.js

```diff
@@ -65,66 +60,61 @@ export default class PolarisBadge extends Component {
    * @default null
    * @public
    */
-  progress = null;
+  progress;
 
   /**
    * Medium or small size. Use `small` only in the main navigation of an app frame..
    *
    * @type {String}
-   * @default 'medium'
+   * @default null
    * @public
    */
-  size = DEFAULT_SIZE;
+  size;
 
   @notEmpty('progress')
   hasProgress;
 
   @notEmpty('status')
   hasStatus;
 
-  @computed('progress')
   get progressDescription() {
-    const { progress } = this;
+    const { progress } = this.args;
```

Comment:
I know this is unchanged, wondering why we have this `default` comparison here.

If it turns out it's not needed, we can likely simplify this a bit, example
```js
  get progressDescription() {
    return this.args.progress ? PROGRESS_LABELS[this.args.progress] : null;
  }
```